### DISCUSSION TITLE
Remove patterns and labels from EntityRuler

### DISCUSSION THREAD

**Discussion by dhruvrnaik** [explosion/spaCy#8228] [New Features & Project Ideas]
I cannot find a good way to remove patterns from a label, or the entire list of patterns corresponding to a label with the EntityRuler pipe.

There is one way by using `entity_ruler.matcher.remove(<LABEL>)`, but this is not a good solution as it doesn't remove labels from `entity_ruler.token_patterns` or `phrase_patterns`.

It would also be great if there was a way to remove a single pattern from a label, and not have to delete the entire list of patterns and the label.

**Comment by polm** [comment]
It's true we don't have a good way to do this. I think the reason is that in our experience it's typical to just build the EntityRuler and use it without modifying it over time. 

I don't think it'd be complicated to add this, but can you clarify how you're using the Entity Ruler that makes this a feature you need? It'll help us understand how to best adjust the API.

**Reply by dhruvrnaik** [reply]
I would want to do entity/phrase detection for different users on a chat platform.

Users would have the flexibility to add and remove phrases they want to detect in real-time chat at any given time. While you can keep adding patterns, and ignore the results for patterns you don't need anymore, I'd prefer having the ability to add/remove patterns based on user needs.

**Reply by polm** [reply]
Thank you, that's a very easy to understand use case and I can see why you'd want to remove patterns in that situation. 

No promises on when it'll get done but we'll consider adding the feature to remove patterns.

As a workaround, I don't think you can do better than directly manipulating the internals or rebuilding the EntityRuler from scratch when things are removed.
